src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -723,6 +781,28 @@ type TableContext<'TRecord> internal
     }
 
 
+    /// <summary>
+    ///     Atomically applies a set of 1-25 write operations to the table.<br/>
+    ///     NOTE requests are charged at twice the normal rate in Write Capacity Units.
+    ///     See the DynamoDB <a href="https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TransactWriteItems.html"><c>TransactWriteItems</c> API documentation</a> for full details of semantics and charges.<br/>
+    /// </summary>
+    /// <param name="items">Operations to be performed.<br/>
+    /// Throws <c>ArgumentOutOfRangeException</c> if item count is not between 1 and 25 as required by underlying API.<br/>
+    /// Use <c>TransactWriteItemsRequest.TransactionCanceledConditionalCheckFailed</c> to identify any Precondition Check failures.</param>
+    /// <param name="clientRequestToken">The <c>ClientRequestToken</c> to supply as an idempotency key (10 minute window).</param>
+    member _.TransactWriteItems(items : seq<TransactWrite<'TRecord>>, ?clientRequestToken) : Async<unit> = async {
+        let reqs = TransactWriteItemsRequest.toTransactItems tableName template items
+        if reqs.Count = 0 || reqs.Count > 25 then raise <| System.ArgumentOutOfRangeException(nameof items, "must be between 1 and 25 items.")
```

Comment:
If you validating items, might as well validate `clientRequestToken` max len.

Reply:
The underlying SDK exception message is pretty ugly, which is why I've opted to guard this case explicitly here. It's also being covered explicitly in the other batch APIs (though the open question in the overview is whether this should match the other messages and/or exn types).

However in general I'm trying to avoid redundantly pre-validating stuff that the underlying SDK validates perfectly well - not least because it then needs to be replicated in the xmldoc etc. (and don't date flag that there's no test using the `cRT` !)

Reply:
The logic in the body of the method immediately passes it to the property, which will throw if it fails validation; leaving it as-is